docs/src/run-with-lambda.md

```diff
@@ -0,0 +1,124 @@
+9. Set “Auth type” to NONE. (Don’t enable CORS, as Martin already will, and you will have duplicate headers and your CORS will be broken.)
+10. Click on the “Function URL”. If it works, hooray! If it doesn’t, open the “Monitor” tab, “View CloudWatch logs”, find the most recent Log stream.
+
+### Zip deployment
+
+It’s possible to deploy the entire codebase from the AWS console, but we will use Serverless Application Model. Our function will consist of a “Layer”, containing the Martin binary, and our function itself will contain the configuration in .yaml format.
+
+#### The layer
+
+Download the binary and place it in your staging directory. The `bin` directory of your Layer will be added to the PATH.
+
+```bash
+mkdir -p martin_layer/src/bin/
+cd martin_layer
+curl -OL https://github.com/maplibre/martin/releases/download/VERSION_NUMBER_HERE/martin-aarch64-unknown-linux-musl.tar.gz
+tar -C src/bin/ -xzf martin-aarch64-unknown-linux-musl.tar.gz martin
+```
+
+Every zip-based Lambda function runs a file called `bootstrap`. `vim src/bootstrap`
+
+```bash
+#!/bin/sh
+exec martin -c ${_HANDLER}.yaml
+```
+
+Write the SAM template. `vim template.yaml`
+
+```yaml
+AWSTemplateFormatVersion: 2010-09-09
+Transform: 'AWS::Serverless-2016-10-31'
+Resources:
+  martin:
+    Type: 'AWS::Serverless::LayerVersion'
+    DeletionPolicy: Delete
+    Properties:
+      ContentUri: src
+      CompatibleRuntimes:
+      - provided.al2023
+      CompatibleArchitectures:
+      - arm64
+Outputs:
+  LayerArn:
+    Value: !Ref MartinLayer
+    Export:
+      Name: !Sub "${AWS::StackName}-LayerArn"
+```
+
+Run `sam deploy --guided`.
+
+1. Stack Name: [martin-layer]
+2. Press enter for everything else
+3. The settings are saved to `samconfig.toml`, so you can later do `sam deploy` to update the version, or `sam delete`.
+
+Now if you visit the [Lambda console](https://console.aws.amazon.com/lambda/home) and select “Layers”, you should see your layer.
+
+#### The function
+
+1. Select “Functions”, “Create function”.
+2. Put something in “Function name”.
+3. Set “Runtime” to “Amazon Linux 2023”.
+4. Set “Architecture” to “arm64”.
+5. Under “Advanced settings”, choose “Enable function URL” with “Auth type” of “NONE”.
+6. Click “Create function”.
+
+Add your layer:
+
+1. Click “add a layer” (green banner at the top, or the very bottom).
+2. Choose “Custom layers”, and select your layer and its version.
+3. Click “Add”.
+
+Add your configuration file in the function source code:
+
+1. Code tab, File, New File: `hello.handler.yaml`.
+
+   ```yaml
+   pmtiles:
+     sources:
+       demotiles: <url to a pmtiles file>
+   ```
+
+2. Click Deploy, wait for the success banner, and visit your function URL.
+
+### Extra things
+
+- Lambda has a default timeout of 3 seconds, and 128 MB of memory, maybe this is suboptimal.
+- Martin will cache tiles in memory, but this might be useless for an ephemeral server.
```

Comment:
I don't think this is useless - if martin binary will handle more than one request, it will be better to cache at least some things - for example, when using pmtiles, cache the directory structure.